src/common/snippets/src/generator.cpp

```diff
@@ -51,9 +52,15 @@ ngraph::snippets::code ngraph::snippets::Generator::generate(std::shared_ptr<ov:
     std::vector<size_t> io_last_dims(in + out);
     std::vector<size_t> io_data_sizes(in + out);
     std::transform(params.begin(), params.end(), io_last_dims.begin(),
-                   [](const std::shared_ptr<Node>& n){return n->get_output_shape(0).back();});
+                   [](const std::shared_ptr<Node>& n){
+                       auto last_dim = n->get_output_partial_shape(0).rbegin();
+                       return last_dim->is_dynamic() ? 0 : last_dim->get_length();
```

Comment:
I'm not sure that dynamic dim should be zero. Because some dimensions can be zeros (for example, after `NonZero`). In plugin for this case we use constant `UNDEFINED_DIMENSION`. Maybe we can use something like that. What do you think?

Reply:
It's a valid concern, thank you. Added `Subgraph::DYNAMIC_DIMENSION`